src/login/components/LoginFormLedger/LoginFormLedger.js

```diff
@@ -56,50 +66,63 @@ export default class LoginFormLedger extends React.Component {
   }
 
   renderActions = () => {
-    const disabled = this.props.disabled || !this.props.deviceInfo;
+    const disabled = (this.state.connected && this.state.open) ? this.props.disabled : true;
     const onClick = disabled ? null : this.handleLogin;
 
+    this.state.deviceError = (!disabled) ?
```

Comment:
This is not a valid manner of changing state within React.  We should only ever use `setState`.  (Please see the [React documentation](https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly) for reference.)

With that said, in line with my previous comment, changing React state during rendering should be avoided at all costs.